lib/Dialect/TT/IR/TTOpsTypes.cpp

```diff
@@ -134,13 +147,72 @@ mlir::tt::SystemDescAttr::getFromPath(MLIRContext *context, std::string &path) {
       break;
     }
 
+    std::vector<tt::DataType> supported_data_types;
```

Comment:
It seems that this variable isn't used.  I think we should move `std::vector<tt::DataTypeAttr> supported_data_types_attr;` up here and use this switch statement below to populate the attributes though.

Reply:
The code that is used doesn't even have the switch statement, so it makes me wonder if it's needed at all.  Agreed above this supported_data_types not being used currently...Maybe try removing the switch statement entirely..?

Reply:
So the code below is just doing `DataType(it)`, which technically is just c-style casting from the flatbuffer enum type to the tablegen enum type.  The only way we can safely do this is if we wrote:
```
static_assert(DataType::Float32 == ::tt::target::DataType::Float32);
static_assert(DataType::Float16 == ::tt::target::DataType::Float16);
....
```

Probably a switch statement is better because then if a new `DataType` is added in the future there will be a compiler error for unhandled case statement.